packages/cadl-ranch-specs/http/type/model/visibility/main.tsp

```diff
@@ -59,23 +82,29 @@ model ReadOnlyModel {
   ```
   """)
 @get
-op getModel(@body input: VisibilityModel): {
+op getModel(@body input: QueryModel): {
   @body
-  output: VisibilityModel;
+  output: ReadModel;
 };
 
 @scenario
 @scenarioDoc("""
-  Generate abd send put model with write/create properties.
-  Expected input body:
+  Generate and receive output model with readonly properties.
+  Expected header: "queryProp": "123"
+  
+  Expected response body:
   ```json
   {
-    queryProp: 123,
+    readProp: "abc",
   }
   ```
   """)
-@head
-op headModel(@body input: VisibilityModel): OkResponse;
+@get
+@route("/getwithquery")
+op getModelWithQuery(@query queryProp: int32): {
```

Comment:
I don't think there's a good reason to remove `headModel`, we can just add `getModelWithQuery` as a separate test

Reply:
I have expanded the types of visibility properties in headmodel and added them back. Perhaps this will be more appropriate.